zerver/views/push_notifications.py

```diff
@@ -254,3 +259,46 @@ def self_hosting_registration_transfer_challenge_verify(
     verification_secret = data["verification_secret"]
 
     return json_success(request, data={"verification_secret": verification_secret})
+
+
+@human_users_only
+@typed_endpoint
+def register_push_device(
+    request: HttpRequest,
+    user_profile: UserProfile,
+    *,
+    token_kind: Annotated[str, check_string_in_validator(PushDevice.TokenKind.values)],
+    push_account_id: Json[int],
+    push_public_key: str,
+    # Key used to encrypt 'encrypted_push_registration'.
+    bouncer_public_key: str,
+    # Registration data encrypted by mobile client for bouncer.
+    encrypted_push_registration: str,
+) -> HttpResponse:
+    # TODO: If push notification not configured, return
+
+    # Idempotency
+    already_registered = PushDevice.objects.filter(
+        user=user_profile, push_account_id=push_account_id, error_code__isnull=True
+    ).exists()
+    if already_registered:
+        return json_success(request)
+
+    PushDevice.objects.update_or_create(
+        user=user_profile,
+        push_account_id=push_account_id,
+        defaults={"token_kind": token_kind, "push_public_key": push_public_key, "error_code": None},
+    )
+
+    queue_event_on_commit(
```

Comment:
I think here we should have a comment like this:

"We used a queue worker to make the request to the bouncer to complete the registration, so that any transient failures can be managed between the two servers, without the mobile device and its often-irregular network access in the picture.".

Reply:
```
    # We use a queue worker to make the request to the bouncer
    # to complete the registration, so that any transient failures
    # can be managed between the two servers, without the mobile
    # device and its often-irregular network access in the picture.
```